components/common/src/main/java/com/hotels/styx/api/ResponseEventListener.java

```diff
@@ -119,19 +83,59 @@ public ResponseEventListener whenFinished(Runnable action) {
     }
 
     public Flux<LiveHttpResponse> apply() {
+
         EventProcessor eventProcessor = new QueueDrainingEventProcessor(
-                new FsmEventProcessor<>(fsm, (throwable, state) -> {
-                }, ""));
+                event -> {
+                    switch (state) {
+                        case INITIAL:
+                            if (event instanceof MessageHeaders) {
+                                state = STREAMING;
+                            } else if (event instanceof MessageCancelled) {
+                                cancelAction.run();
+                                whenFinishedAction.run();
+                                state = TERMINATED;
+                            } else if (event instanceof MessageCompleted) {
+                                // TODO: Add custom exception type?
```

Comment:
Should we deal with this TODO / remove it?  What do you think of dating TODO comments?

Reply:
Hi David. This is an old TODO that appears as a "new" feature in the diffs.

A date could be helpful for code reviewers to spot old TODO items that have merely moved around. But I'm not opinionated about this.